### DISCUSSION TITLE
[Flows] access `process.env` in Run Script operation

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#16108] [Feature Requests]
hi:

I tried access `process.env.something`, looks like the variable `process.env` always empty object, how about this.

1. I thinking access env available have not any security risk.

**Comment by rijkvanzanten** [comment]
> I thinking access env available have not any security risk.

I strongly disagree. App users aren't necessarily server admins, and `process.env` allows access to _all_ environment variables, including aws injected access keys, database passwords, etc etc

**Reply by imiskolee** [reply]
how about with prefix  `DIRECTUS_CUSTOM_`

**Reply by janhartje** [reply]
I like the Idea it should help to keep Flows consistant on multiple environments. The Idea with the prefix should could solve it too.

**Reply by imiskolee** [reply]
> I like the Idea it should help to keep Flows consistant on multiple environments. The Idea with the prefix should could solve it too.

yes. i am also faced to multiple environments problem.

**Reply by licitdev** [reply]
PR #16111 enables access to the environment variables in the Flows Run Script operation.

With regards to the prefix idea, kindly discuss it in #13348 instead. Thank you.

**Comment by nickrum** [comment]
Env access in Run Script operation has been added in #16111.
